### PR TITLE
DDP-6330. Add A-T DSM custom workflow status events

### PR DESCRIPTION
**Pull request by ssettipalli**
look at https://broadinstitute.atlassian.net/browse/DDP-6330
Add needed A-T DSM custom workflow status events

### CODE REVIEW

File: study-builder/studies/atcp/study.conf [hunk 1]
```diff
@@ -1239,6 +1239,180 @@
...
+        "type": "ACTIVITY_STATUS",
+        "activityCode": "REGISTRATION",
+        "statusType": "CREATED"
+      },
+      "action": {
+        "type": "UPDATE_CUSTOM_WORKFLOW",
+        "workflow": "REGISTRATION_TYPE",
+        "status": "Self"
+      },
+      "preconditionExpr": """operator.studies["atcp"].forms["PREQUAL"].questions["PREQUAL_SELF_DESCRIBE"].answers.hasOption("DIAGNOSED")"""
```
Comment (yufengwng) [inline study-builder/studies/atcp/study.conf:1380]: Are we sure it's the `operator` and not the participant `user`?

File: study-builder/studies/atcp/study.conf [hunk 2]
```diff
@@ -1239,6 +1239,180 @@
...
+        "type": "ACTIVITY_STATUS",
+        "activityCode": "REGISTRATION",
+        "statusType": "COMPLETE"
+      },
+      "action": {
+        "type": "UPDATE_CUSTOM_WORKFLOW",
+        "workflow": "REGISTRATION_TYPE",
+        "status": "Dependent"
+      },
+      "preconditionExpr": """operator.studies["atcp"].forms["PREQUAL"].questions["PREQUAL_SELF_DESCRIBE"].answers.hasOption("CHILD_DIAGNOSED")"""
```
Comment (yufengwng) [inline study-builder/studies/atcp/study.conf:1412]: This is the same expression as the `PortalUser` one above?

Comment (ssettipalli) [reply study-builder/studies/atcp/study.conf:1412]: Expr is same but trigger is after REGISTRATION COMPLETE

File: study-builder/studies/atcp/study.conf (hunk 1 above)
Comment (ssettipalli) [reply study-builder/studies/atcp/study.conf:1380]: looks like A-T uses operator for all prequal related PEX checks

**Review by gmakharat — APPROVED** [review]

**Review by yufengwng — APPROVED** [review]

**Pull request merged**